controllers/snapshot/snapshot_adapter.go

```diff
@@ -366,6 +369,29 @@ func (a *Adapter) EnsureGlobalCandidateImageUpdated() (controller.OperationResul
 	return controller.ContinueProcessing()
 }
 
+func GetStatusCondition(snapshot *applicationapiv1alpha1.Snapshot) (condition *metav1.Condition, ok bool) {
```

Comment:
Thinking if this function would make more sense being placed in [gitops/snapshot.go](https://github.com/redhat-appstudio/integration-service/blob/main/gitops/snapshot.go)?

Reply:
It initially started out life in gitops/snapshot.go. Current change is based on feedback from mbasti on that code, feedback:

1. in metrics in RegisterReleaseLatency just use startTime metav1.Time as parameter (keep endCreateRelase as time.Now()), and define function that extracts startTime directly in snapshot controller adapter
2. you have to write new function to get data, confition, ok := getStatusCondition(snapshot) or smth like that, and also keep LegacyTestsSucceededCondition there

Its possible that number 2 was to remain in gitops/snapshot.go and I misunderstood. i will have a look at that and see how it might work.

Reply:
works for me to have it in gitops/snapshot